core/frontend/src/components/wifi/WifiUpdater.vue

```diff
@@ -0,0 +1,120 @@
+            return
+          }
+
+          const scanned_network = wifi_store.available_networks.find(
+            (network) => network.ssid === response.data.ssid,
+          )
+          const saved_network = wifi_store.saved_networks.find(
+            (network) => network.ssid === response.data.ssid,
+          )
+
+          wifi_store.setCurrentNetwork({
+            ssid: response.data.ssid,
+            signal: scanned_network ? scanned_network.signal : 0,
+            locked: response.data.key_mgmt.includes('WPA'),
+            saved: saved_network != null,
+          })
+        })
+        .catch((error) => {
+          common_store.addError(new WifiStatusError(`Could not fetch wifi status: ${error.message}`))
+          wifi_store.setCurrentNetwork(null)
+        })
+    },
+    async fetchAvailableNetworks (): Promise<void> {
+      await axios({
+        method: 'get',
+        url: `${wifi_store.API_URL}/scan`,
+        timeout: 20000,
+      })
+        .then((response) => {
+          common_store.removeError(ErrorId.WifiScanError)
+
+          const saved_networks_ssids = wifi_store.saved_networks.map((network: SavedNetwork) => network['ssid'])
+          const available_networks = response.data.map((network: WPANetwork) => {
+            return {
+              ssid: network.ssid,
+              signal: network.signallevel,
+              locked: network.flags.includes('WPA'),
+              saved: saved_networks_ssids.includes(network.ssid),
+            }
+          })
+          wifi_store.setAvailableNetworks(available_networks)
+        })
+        .catch((error) => {
+          common_store.addError(new WifiScanError(`Could not scan for wifi networks: ${error.message}`))
+          wifi_store.setAvailableNetworks([])
+        })
+    },
+    async fetchSavedNetworks (): Promise<void> {
+      await axios({
+        method: 'get',
+        url: `${wifi_store.API_URL}/saved`,
+        timeout: 2000,
+      })
+        .then((response) => {
+          common_store.removeError(ErrorId.WifiSavedError)
+          wifi_store.setSavedNetworks(response.data)
+        })
+        .catch((error) => {
+          common_store.addError(new WifiSavedError(`Could not fetch saved networks: ${error.message}.`))
+          wifi_store.setSavedNetworks([])
+        })
+    },
+    sleep (ms: number): Promise<void> {
+      return new Promise((resolve) => setTimeout(resolve, ms))
+    },
+    async startSavedNetworksUpdateLoop (): Promise<void> {
+      await this.fetchSavedNetworks()
+      await this.sleep(5000)
+      this.startSavedNetworksUpdateLoop()
+    },
+    async startNetworkStatusUpdateLoop (): Promise<void> {
+      await this.fetchNetworkStatus()
+      await this.sleep(5000)
+      this.startNetworkStatusUpdateLoop()
+    },
+    async startNetworksScanLoop (): Promise<void> {
+      await this.fetchAvailableNetworks()
+      await this.sleep(5000)
+      this.startNetworksScanLoop()
```

Comment:
We could abstract this in a better way.  maybe a periodicRequest function ?

Reply:
Good idea!